### DISCUSSION TITLE
font file issues - naming, directory structure

### DISCUSSION THREAD

**Discussion by nbriggs** [Interlisp/medley#53] [configurations and releases]
Code for searching for font files needs to be compatible with the directory structure the font files are stored in.
Currently you'd expect DISPLAYFONTDIRECTORIES to be set to a small number of locations for display fonts.
The font searching code should find the files under that directory.  Unfortunately, the directory structure seems to include an extra level to split out the files by XCCS character set number (e.g., C0), and the font search code isn't taking this into account, so you get a search like:

```
1 - Enter \READDISPLAYFONTFILE:
          FAMILY = HIPPO
          SIZE = 8
          FACE = (MEDIUM REGULAR REGULAR)
          ROTATION = 0
          DEVICE = DISPLAY
          CHARSET = 0
    2 - Enter \FINDFONTFILE:
              FAMILY = HIPPO
              SIZE = 8
              FACE = (MEDIUM REGULAR REGULAR)
              ROTATION = 0
              DEVICE = DISPLAY
              CHARSET = 0
              DIRLST = ("{DSK}/Users/briggs/Projects/fonts/displayfonts/")
              EXTLST = (DISPLAYFONT)
        3 - Enter FINDFILE:
                  FILE = "HIPPO08-MRR-C0.DISPLAYFONT"
                  NSFLG = T
                  DIRLST = ("{DSK}/Users/briggs/Projects/fonts/displayfonts/")
        3 - Exit FINDFILE =>
                  NIL
        3 - Enter FINDFILE:
                  FILE = "HP08R0.DISPLAYFONT"
                  NSFLG = T
                  DIRLST = ("{DSK}/Users/briggs/Projects/fonts/displayfonts/")
        3 - Exit FINDFILE =>
                  NIL
    2 - Exit \FINDFONTFILE =>
         NIL
```
so the font file was not found, though it was stored in displayfonts/C0/HIPPO08-MRR-C0.DISPLAYFONT.
Also, if you're running on a case-sensitive file system, the font files need to be in uppercase because that's the way the names will be constructed in the Lisp code.

**Comment by masinter** [comment]
I think I was confused about maiko and case insensitive filenames. But in this case, there's an explicit comment
https://gyazo.com/ed13ec37bde840cc8e5d35eae000bcbf
in \FONTFILENAME which prepends a "c0>" (lower case) if *USEOLDFONTDIRECTORIES* is NIL.

**Comment by nbriggs** [comment]
Ah, it's an inconsistency between the sysouts that I have -- my lisp.sysout and full.sysout have FONT.;2 while xfull35 has FONT.;4 and the <code>\*USEOLDFONTDIRECTORIES\*</code> was introduced in FONT.;3

**Comment by masinter** [comment]
the font code is a complex maze of guesswork and flgs meant to optimize stuff that no longer matters, (While Lisp code runs 200 times faster, file systems are a lot faster than NFS on a Sun ).
